lib/tesla/adapter/hackney.ex

```diff
@@ -72,8 +72,18 @@ if Code.ensure_loaded?(:hackney) do
 
     defp request_stream(method, url, headers, body, opts) do
       with {:ok, ref} <- :hackney.request(method, url, headers, :stream, opts) do
-        for data <- body, do: :ok = :hackney.send_body(ref, data)
-        handle(:hackney.start_response(ref))
+        res =
+          Enum.reduce_while(body, :ok, fn data, _ ->
+            case :hackney.send_body(ref, data) do
+              :ok -> {:cont, :ok}
+              error -> {:halt, error}
+            end
+          end)
+
+        case res do
+          :ok -> handle(:hackney.start_response(ref))
+          error -> handle(error)
+        end
```

Comment:
We could extract sending into separate function and then use it like:

```elixir
with {:ok, ref} <- :hackney.request(method, url, headers, :stream, opts) do
    :ok <- send_body(ref, body)
  handle(:hackney.start_response(ref))
else
  e -> handle(e)
end
```

Reply:
Will extract it to function then :)
But I think `with` won't work here as we return accumulator (this time only :ok) or error.